managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorTest.java

```diff
@@ -3750,5 +3750,25 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
         Awaitility.await().untilAsserted(() -> assertTrue(flag.get()));
     }
 
+    @Test
+    public void testLazyCursorLedgerCreation() throws ManagedLedgerException, InterruptedException {
+        ManagedLedgerConfig managedLedgerConfig = new ManagedLedgerConfig();
+        ManagedLedgerImpl ledger = (ManagedLedgerImpl) factory
+                .open("testLazyCursorLedgerCreation", managedLedgerConfig);
+        ManagedCursorImpl cursor = (ManagedCursorImpl) ledger.openCursor("test");
+        assertEquals(cursor.getState(), "NoLedger");
```

Comment:
We should also assert the mark-delete position of the cursor before any ack is sent. 

And we could close the cursor (either gracefully or not) and recover it to verify the mark-delete is correct.

To simulate the "ungraceful" close, we can use a different ManagedLedgerFactory to fence and recover the managed ledger.